lib/ASN1/Universal/Integer.php

```diff
@@ -60,14 +65,13 @@ protected function getEncodedValue()
         $contentLength = $this->getContentLength();
 
         if ($numericValue < 0) {
```

Comment:
Will this check work if `$this->value` is a huge negative integer?

Maybe it would be better to initialize `$this->value` in the constructor and from then on always rely on it being a gmp value and consequently apply the according gmp operations. What do you think?